index.js

```diff
@@ -0,0 +1,170 @@
+    let grade = document.querySelector("#grade").value;
+    let container = document.querySelector("#addStudentDiv");
+    // let cnp = document.createElement("p");
+    let snp = document.createElement("p");
+    let ap = document.createElement("p");
+    let cp = document.createElement("p");
+    let gp = document.createElement("p");
+    // cnp.innerText = className;
+    snp.innerText = studentName;
+    ap.innerText = age;
+    cp.innerText = city;
+    gp.innerText = grade;
+    container.append(snp, ap, cp, gp);
+    
+console.log('in the formjhfjyfjytf');
+
+
+    try {
+        console.log('in the try', className);
+        
+        let res = await axios.post(`http://localhost:3000/class/${className}/enroll`, {
+            className,
+            studentName,
+            age,
+            city,
+            grade
+        })
+        console.log('adding student',res);
+        
+        let p = document.createElement("p");
+        p.innerText = JSON.stringify(res.data);
+        container.appendChild(p);
+    } catch (err) {
+        console.log('in the catch');
+        // debugger
+        console.log(err)
+    }
+})
+
+
+const populateSelect = (className) => {
+    let option = document.createElement("option");
+    option.innerText = className;
+    console.log('new class option=>', option )
+    select.appendChild(option);
+}
+
+const addClass = async () => {
+    try {
+        let name = document.querySelector("#className").value;
+        let teacher = document.querySelector("#teacherName").value;
+        let res = await axios.post("http://localhost:3000/class", {
+            name,
+            teacher
+        });  
+        console.log('THis is res=> ',res)
+        populateSelect(name);
+        return res; 
+    } catch (error) {
+        console.log(error);
+    }
+}
+
+const addClassToDom = async () => {
+    data = await addClass();
+    console.log(data);
+    
+    let container = document.querySelector("#addClassDiv");
+
+    let newClass = document.createElement("div");
+    newClass.id = 'newClass';
+    let className = document.createElement("p");
+    let teacherName = document.createElement("p");
+    
+    className.innerText = data.data.class.name;
+    teacherName.innerText = data.data.class.teacher;
+
+    newClass.append(className,teacherName);
+
+    container.append(newClass);
+}
+
+const checkBoxChange = async (check) => {
+    let classList = document.querySelector("#classList").value;
+    let cityList = document.querySelector("#cityList").value;
+        // console.log(check.checked);
+        console.log(check)
+        try {
+            let res = await axios.get(`http://localhost:3000/class/${classList}/students/?failing=${check}&city=${cityList}`);
+            console.log(res);
+            console.log('this is allClasses', res.data.allClasses);
+            console.log(res);
+        
+            let result = [];
+            let allClasses = res.data.allClasses[classList]; 
```

Comment:
Your response from the api is not being properly handled here. Im making a request it should give me an error. But since its not throwing the error its trying to loop through an array that doesn't exist.